x509/src/certificate.rs

```diff
@@ -202,6 +202,33 @@ pub struct Certificate<'a> {
     pub signature: BitString<'a>,
 }
 
+impl<'a> DecodeValue<'a> for Certificate<'a> {
+    fn decode_value(decoder: &mut Decoder<'a>, _length: Length) -> der::Result<Self> {
+        let tbs_certificate = decoder.decode()?;
+        let signature_algorithm = decoder.decode()?;
+        let signature = decoder.decode()?;
+        Ok(Self {
+            tbs_certificate,
+            signature_algorithm,
+            signature,
+        })
+        // })
+    }
+}
+
+impl<'a> ::der::Sequence<'a> for Certificate<'a> {
+    fn fields<F, T>(&self, f: F) -> ::der::Result<T>
+    where
+        F: FnOnce(&[&dyn der::Encodable]) -> ::der::Result<T>,
+    {
+        f(&[
+            &self.tbs_certificate,
+            &self.signature_algorithm,
+            &self.signature,
+        ])
+    }
+}
```

Comment:
Why doesn't custom derive work here?

Reply:
The tagged field in CertPathControls generated that need. Same thing for NameConstraints (sibling tagged field in CertPathControls). Likely the same issue underpins the OCSP issue I mentioned earlier today as well (where GeneralName is the target).

Reply:
If you put Certificate, NameConstraints and CertPathControls back with Sequence derive and corresponding context-specific attributes (which is how they started), the following error results:

```
error[E0277]: the trait bound `certificate::Certificate<'_>: DecodeValue<'_>` is not satisfied
   --> x509/src/trust_anchor_format.rs:146:39
    |
146 | #[derive(Clone, Debug, Eq, PartialEq, Sequence)]
    |                                       ^^^^^^^^ the trait `DecodeValue<'_>` is not implemented for `certificate::Certificate<'_>`
    |
note: required by a bound in `der::asn1::ContextSpecific::<T>::decode_implicit`
   --> der/src/asn1/context_specific.rs:71:12
    |
71  |         T: DecodeValue<'a> + Tagged,
    |            ^^^^^^^^^^^^^^^ required by this bound in `der::asn1::ContextSpecific::<T>::decode_implicit`
    = note: this error originates in the derive macro `Sequence` (in Nightly builds, run with -Z macro-backtrace for more info)
```
```
error[E0277]: the trait bound `pkix_extensions::NameConstraints<'_>: DecodeValue<'_>` is not satisfied
   --> x509/src/trust_anchor_format.rs:146:39
    |
146 | #[derive(Clone, Debug, Eq, PartialEq, Sequence)]
    |                                       ^^^^^^^^ the trait `DecodeValue<'_>` is not implemented for `pkix_extensions::NameConstraints<'_>`
    |
note: required by a bound in `der::asn1::ContextSpecific::<T>::decode_implicit`
   --> der/src/asn1/context_specific.rs:71:12
    |
71  |         T: DecodeValue<'a> + Tagged,
    |            ^^^^^^^^^^^^^^^ required by this bound in `der::asn1::ContextSpecific::<T>::decode_implicit`
    = note: this error originates in the derive macro `Sequence` (in Nightly builds, run with -Z macro-backtrace for more info)

```

Reply:
It worked and changes were pushed to catch up #356.